kalite/version.py

```diff
@@ -5,10 +5,15 @@
 # THIS IS USED BY settings.py.  NEVER import settings.py here; hard-codes only!
 # Must be PEP 440 compliant: https://www.python.org/dev/peps/pep-0440/
 # Must also be of the form N.N.N for internal use, where N is a non-negative integer
-MAJOR_VERSION = "0"
-MINOR_VERSION = "14"
-PATCH_VERSION = "0"
-VERSION = "%s.%s.%s" % (MAJOR_VERSION, MINOR_VERSION, PATCH_VERSION)
+
+from django.utils.version import get_version
+
+VERSION_TUPLE = (0, 14, 0, "alpha", 0)
+
+MAJOR_VERSION, MINOR_VERSION, PATCH_VERSION = map(str, VERSION_TUPLE[:3])
+
+VERSION = get_version(VERSION_TUPLE)
```

Comment:
@benjaoming @mcgallaspy I spent some time digging into this on the airplane, and I'm glad I did!

My concern was that the function `fle_utils.general.version_diff` (which gets used for minversion comparison for critical securesync functions, in `fle_utils.django_utils.serializers.versioned_json.Serializer`) barfs if there are non-integer components passed in as part of the version string. However, it is (meant to be; see below) using `kalite.version.VERSION`, which only includes the first three components, so only the first three components need to be integers. Hence, `(0, 14, 0, "alpha", 42)` would be OK, but `(0, 14, "alpha", 42)` or `(0, 14, "0alpha")` etc would NOT be.

However, in looking into this, I discovered that as of 0.14, the proper `VERSION` is no longer being used by securesync, with [this commit](https://github.com/learningequality/ka-lite/commit/66f996e1d3157201dcdf2b8984be51eb13f77bf1#diff-2fe3caeae433a31561ccb9c8334aaa92R15) in April (as part of the settings refactor, I believe; @benjaoming), and now `securesync.VERSION` is always hard-coded to "1.0". Since "1.0" is bigger than all our `minversion` values, we would never exclude any models/fields when syncing with older versions, leading to bad sync errors.

I put a fix in at https://github.com/learningequality/ka-lite/pull/4205